Documentation Standard Updates (#51)
fivetran/dbt_mixpanel: README.md

</p>

# Mixpanel dbt Package ([Docs](https://fivetran.github.io/dbt_mixpanel/))
## What does this dbt package do?

- Produces modeled tables that leverage Mixpanel data from [Fivetran's connector](https://fivetran.com/docs/applications/mixpanel). It uses the Mixpanel `event` table in the format described by [this ERD](https://fivetran.com/docs/applications/mixpanel#schemainformation).

- Enables you to better understand user activity and retention through your event data. It:

> - Creates both a daily and monthly timeline of each type of event, complete with metrics about user activity, retention, resurrection, and churn
> - Aggregates events into unique user sessions, complete with metrics about event frequency and any relevant fields from the session's first event
> - Provides a macro to easily create an event funnel
> - De-duplicates events according to [best practices from Mixpanel](https://developer.mixpanel.com/reference/http#event-deduplication)
> - Pivots out custom event properties from JSONs into an enriched events table

Author: ```suggestion
  - Creates both a daily and monthly timeline of each type of event, complete with metrics about user activity, retention, resurrection, and churn
  - Aggregates events into unique user sessions, complete with metrics about event frequency and any relevant fields from the session's first event
  - Provides a macro to easily create an event funnel
  - De-duplicates events according to [best practices from Mixpanel](https://developer.mixpanel.com/reference/http#event-deduplication)
  - Pivots out custom event properties from JSONs into an enriched events table
```

<!--section="mixpanel_transformation_model-->
- Generates a comprehensive data dictionary of your source and modeled Mixpanel data through the [dbt docs site](https://fivetran.github.io/dbt_mixpanel/#!/overview).

> The following table provides a detailed list of all models materialized within this package by default.
> > TIP: See more details about these models in the package's [dbt docs site](https://fivetran.github.io/dbt_mixpanel/#!/overview?g_v=1).
> 
> | **Model**                | **Description**                                                                                                                                |

Author: ```suggestion
The following table provides a detailed list of all tables materialized within this package by default.
> TIP: See more details about these tables in the package's [dbt docs site](https://fivetran.github.io/dbt_mixpanel/#!/overview?g_v=1).

| **Table**                | **Description**                                                                                                                                |
```

    mixpanel_schema: your_schema_name 
```

### (Optional) Step 4: Additional configurations
<details open><summary>Collapse/expand details</summary>

### Macros
#### analyze_funnel [(source)](https://github.com/fivetran/dbt_mixpanel/blob/master/macros/analyze_funnel.sql)
You can use the `analyze_funnel(event_funnel, group_by_column, conversion_criteria)` macro to produce a funnel between a given list of event types.

It returns the following:
- The number of events and users at each step
- The overall user and event conversion % between the top of the funnel and each step
- The relative user and event conversion % between subsequent steps
> Note: The relative order of the steps is determined by their event volume, not the order in which they are input.

The macro takes the following as arguments:
- `event_funnel`: List of event types (not case sensitive).

> - Example: `'['play_song', 'stop_song', 'exit']`
> - `group_by_column`: (Optional) A column by which you want to segment the funnel (this macro pulls data from the `mixpanel__event` model). The default value is `None`.
> - Example: `group_by_column = 'country_code'`.
> - `conversion_criteria`: (Optional) A `WHERE` clause that will be applied when selecting from `mixpanel__event`.
> - Example: To limit all events in the funnel to the United States, you'd provide `conversion_criteria = 'country_code = "US"'`. To limit the events to only song play events to the US, you'd input `conversion_criteria = 'country_code = "US"' OR event_type != 'play_song'`.

Author: ```suggestion
  - Example: `'['play_song', 'stop_song', 'exit']`
- `group_by_column`: (Optional) A column by which you want to segment the funnel (this macro pulls data from the `mixpanel__event` model). The default value is `None`.
  - Example: `group_by_column = 'country_code'`.
- `conversion_criteria`: (Optional) A `WHERE` clause that will be applied when selecting from `mixpanel__event`.
  - Example: To limit all events in the funnel to the United States, you'd provide `conversion_criteria = 'country_code = "US"'`. To limit the events to only song play events to the US, you'd input `conversion_criteria = 'country_code = "US"' OR event_type != 'play_song'`.
```
